### PR TITLE
Additional option parameters for apoc.index.addAllNodes

### PR DESCRIPTION
**Pull request by dkrizic**
e.g.

`CALL apoc.index.addAllNodes('name',{ Supplier: ["companyName", "description","address"], Product:  ["productName","address"] }, {to_lower_case: false, analyzer: "org.apache.lucene.analysis.core.WhitespaceAnalyzer", autoUpdate: true, queue: "asciifolding" } )`

Next steps: Add to trigger if autoUpdate is true, @sarmbruster had some ideas about this queue stuff.

### CODE REVIEW

**Comment by jexp** [comment]
Looks good to me. I'd like to ask you for 2 things:

1. as this changes/breaks the API we need to change the method name (for now until we have 3.1 with default parameter values) to e.g. `addAllNodesWithConfig` which should be deprecated immediately and used with default param in the 3.1 branch
2. squash your commits (but I can do that too)

**Comment by dkrizic** [comment]
@jexp Please take a look now, obviously my new commits are added to new existing Pull Request.

File: src/main/java/apoc/index/FreeTextSearch.java [hunk 1]
```diff
@@ -57,12 +57,32 @@ private IndexStats(String label, String property, long nodeCount) {
     @Procedure
     @PerformsWrites
     @Description("apoc.index.addAllNodes('name',{label1:['prop1',...],...}) YIELD type, name, config - create a free text search index")
-    public Stream<IndexStats> addAllNodes(@Name("index") String index, @Name("structure") Map<String, List<String>> structure) {
+    @Deprecated
```
Comment (jexp) [inline src/main/java/apoc/index/FreeTextSearch.java]: actually the new method should be deprecated, as in 3.1. we'll have default parameters, and don't need the extra name anymore. :)

File: src/main/java/apoc/index/FreeTextSearch.java [hunk 2]
```diff
@@ -176,9 +196,18 @@ private void populate(Index<Node> index, Map<String, List<String>> config, Consu
...
+
+            /* add options to the parameters */
+            for( String key : options.keySet() ) {
+                config.put( key, "" + options.get( key ) ); // explicit conversion to String
+            }
+            log.info("Creating or updating index '%s' with config '%s'", index, config );
             Index<Node> nodeIndex = db.index().forNodes(index, config);
             tx.success();
             return nodeIndex;
+        } catch( Exception e ) {
```
Comment (jexp) [inline src/main/java/apoc/index/FreeTextSearch.java]: should we really hide exceptions?

File: src/main/java/apoc/index/FreeTextSearch.java [hunk 3]
```diff
@@ -176,9 +196,18 @@ private void populate(Index<Node> index, Map<String, List<String>> config, Consu
         }
         try (Transaction tx = db.beginTx()) {
             updateConfigFromParameters(config, structure);
+
+            /* add options to the parameters */
+            for( String key : options.keySet() ) {
```
Comment (jexp) [inline src/main/java/apoc/index/FreeTextSearch.java]: options.forEach() or entrySet()
probably rather use value.toString()

File: src/main/java/apoc/index/FreeTextSearch.java (hunk 1 above)
Comment (dkrizic) [inline src/main/java/apoc/index/FreeTextSearch.java]: Ok

File: src/main/java/apoc/index/FreeTextSearch.java (hunk 2 above)
Comment (dkrizic) [inline src/main/java/apoc/index/FreeTextSearch.java]: Oops, was for testing

File: src/main/java/apoc/index/FreeTextSearch.java (hunk 3 above)
Comment (dkrizic) [inline src/main/java/apoc/index/FreeTextSearch.java]: Ok

**Comment by dkrizic** [comment]
Can you take a look now, please?

File: src/main/java/apoc/index/FreeTextSearch.java [hunk 4]
```diff
@@ -57,12 +57,34 @@ private IndexStats(String label, String property, long nodeCount) {
     @Procedure
     @PerformsWrites
     @Description("apoc.index.addAllNodes('name',{label1:['prop1',...],...}) YIELD type, name, config - create a free text search index")
-    public Stream<IndexStats> addAllNodes(@Name("index") String index, @Name("structure") Map<String, List<String>> structure) {
+    public Stream<IndexStats> addAllNodes(@Name("index") String index, @Name("structure") Map<String, List<String>> structure ) {
+        return addAllNodesExtended( index, structure, new HashMap<String,Object>(0) );
```
Comment (jexp) [inline src/main/java/apoc/index/FreeTextSearch.java]: Collections.emptyMap()

**Pull request merged**